Mathlib/Analysis/Convex/Radon.lean

```diff
@@ -48,3 +58,126 @@ theorem radon_partition (h : ¬ AffineIndependent 𝕜 f) :
+        · intro J hJ_card
+          rw [show ⋂ i ∈ J, F' i = ⋂ i ∈ Finset.image Subtype.val J, F i by
+            simp [Set.iInter_subtype]]
+          exact h_inter _ (le_trans Finset.card_image_le hJ_card)
+        · simp [ι']; omega
+
+      /- This family of vectors is not affine independent because the number of them exceeds the
+      dimension of the space. -/
+      have h_ind : ¬AffineIndependent 𝕜 a := by
+        rw [← finrank_vectorSpan_le_iff_not_affineIndependent 𝕜 a (n := (k - 1))]
+        · exact (Submodule.finrank_le (vectorSpan 𝕜 (Set.range a))).trans (Nat.le_pred_of_lt h_card)
+        · omega
+      /- Use `Convex.radon_partition` to conclude there is a subset `I` of `ι` and a point `p : E`
+      which lies in the convex hull of either `a '' I` or `a '' Iᶜ`. We claim that `p ∈ ⋂₀ F`. -/
+      obtain ⟨I, p, h4_I, h4_Ic⟩ := Convex.radon_partition h_ind
+      use p
+      apply Set.mem_iInter_of_mem
+      intro i
+      /- It suffices to show that for any set `F i` in a subcollection `I` of `ι`, that the convex
+      hull of `a '' Iᶜ` is contained in `F i`. -/
+      suffices ∀ I : Set ι, i ∈ I → (convexHull 𝕜) (a '' Iᶜ) ⊆ F i by
+        by_cases h : i ∈ I
+        · exact this I h h4_Ic
+        · apply this Iᶜ h; rwa [compl_compl]
+      /- Given any subcollection `I` of `ι` containing `i`, because `F i` is convex, we need only
+      show that `a j ∈ F i` for each `j ∈ Iᶜ`. -/
+      intro I hi
+      rw [Convex.convexHull_subset_iff (h_convex i)]
+      rintro - ⟨j, hj, rfl⟩
+      /- Since `j ∈ Iᶜ` and `i ∈ I`, we conclude that `i ≠ j`, and hence by the definition of `a`:
+      `a j ∈ ⋂ F '' (Set.univ \ {j}) ⊆ F i`. -/
+      apply Set.mem_of_subset_of_mem (s₁ := ⋂₀ (F '' (Finset.univ.erase j)))
+      · apply sInter_subset_of_mem
+        use i
+        simp only [coe_erase, coe_univ, mem_diff, Set.mem_univ, mem_singleton_iff, true_and,
+          and_true]
+        exact fun h' ↦ hj (h' ▸ hi)
+      · apply Set.Nonempty.some_mem
+
+/-- The set version of `Convex.helly_theorem`. -/
+theorem helly_theorem_set {F : Finset (Set E)}
+    (h_convex : ∀ X ∈ F, Convex 𝕜 X)
+    (h_inter : ∀ G : Finset (Set E), G ⊆ F → G.card ≤ finrank 𝕜 E + 1 → (⋂₀ G : Set E).Nonempty) :
+    (⋂₀ (F : Set (Set E))).Nonempty := by
+  classical -- for DecidableEq, required for the family version
+  rw [show ⋂₀ F = ⋂ X : F, (X : Set E) by ext; simp]
+  apply Convex.helly_theorem (F := fun x : F ↦ x.val) (𝕜 := 𝕜)
+  · exact fun X ↦ h_convex X (by simp)
+  · intro G _
+    let G' : Finset (Set E) := Finset.image Subtype.val G
+    rw [show ⋂ i ∈ G, ↑i = ⋂₀ (G' : Set (Set E)) by simp [G']]
+    apply h_inter G'
+    · rw [Finset.image_subset_iff]
+      simp
+    · apply le_trans Finset.card_image_le
+      assumption
+
+/-- The version of `Convex.helly_theorem` that includes an additional compactness assumption, and is
+not restricted to finite families. -/
+theorem helly_theorem_infinite [TopologicalSpace E] [T2Space E] (F : ι → Set E)
```

Comment:
Technically, this does not ask the family to be infinite, so I would rather call it
```suggestion
theorem helly_theorem_family_compact [TopologicalSpace E] [T2Space E] (F : ι → Set E)
```
and note that you are missing `helly_theorem_set_compact`, the set version